internal/tasks/consumer.go

```diff
@@ -6,14 +6,15 @@ import (
 	"fmt"
 
 	"github.com/flightctl/flightctl/internal/store"
+	"github.com/flightctl/flightctl/internal/crypto"
 	"github.com/flightctl/flightctl/pkg/k8sclient"
 	"github.com/flightctl/flightctl/pkg/queues"
 	"github.com/sirupsen/logrus"
 )
 
 const TaskQueue = "task-queue"
 
-func dispatchTasks(store store.Store, callbackManager CallbackManager, k8sClient k8sclient.K8SClient, configStorage ConfigStorage) queues.ConsumeHandler {
+func dispatchTasks(store store.Store, callbackManager CallbackManager, k8sClient k8sclient.K8SClient, configStorage ConfigStorage, ca crypto.CA) queues.ConsumeHandler {
```

Comment:
@avishayt, @ori-amizur is this how developers should extend tasks by extending the signature? The result is folks creating globals to work around it.